packages/core/streams/parsers.ts

```diff
@@ -0,0 +1,55 @@
+import { AIStreamMode, AIStreamParser } from './types'
+import { trimStartOfStreamHelper } from './utils'
+
+export type AIStreamParsers = () => {
+  [mode in AIStreamMode]?: AIStreamParser
+}
+
+export const openaiParsers: AIStreamParsers = () => {
+  const trimStartOfStream = trimStartOfStreamHelper()
+
+  return {
+    raw: undefined,
+    text: data => {
+      // TODO: Needs a type
+      const json = JSON.parse(data)
+
+      // this can be used for either chat or completion models
+      const text = trimStartOfStream(
+        json.choices[0]?.delta?.content ?? json.choices[0]?.text ?? ''
+      )
+
+      return text
+    }
+  }
+}
```

Comment:
We can easily add JSON parsing on completions for integration with [OpenAI function calling](https://openai.com/blog/function-calling-and-other-api-updates) (or manual equivalents, e.g. `Respond only with valid JSON`-like prompts) by updating `type AIStreamMode = "raw" | "text" | "json"` and adding the OpenAI and Anthropic parsers.

Reply:
Check out the long block comment I've got in my PR (#154 ). The way OpenAI streams function calling responses is a little weird.

Reply:
Very weird, I haven't gotten hands-on with it yet. I will look at this more in depth tomorrow, TY for note. 

Sounds like it's probably better to merge yours first, and then resolve conflicts here before this one ships, if anything.